### ISSUE TITLE
TiVo App Won't Stream

### ISSUE THREAD

**Issue by imatechnerd** [n00neimp0rtant/xCon-Issues#630]
I'm stuck using TiVo's iOS app version 3.5.3 b/c the newer versions say "unauthorized modifications detected" despite having xCon installed. Any ideas? Thanks!

**Comment by ContextPerspective** [comment]
Close all apps. In terminal run this...

ldid -S /Library/MobileSubstrate/DynamicLibraries/xCon.dylib

Exit. Reboot. Should solve the issue.

**Comment by Gigglequad** [comment]
I have the same issue (Tivo not streaming) however I'm using v3.7.3 of the app and v42b2 of xcon. The above command didn't make a difference. Any other ideas? IOS is 7.1.2.

**Comment by ContextPerspective** [comment]
What happens? Are you getting the modifications error or something else? I want to say that tivo dropped support for iOS 7 in the latest stream software, buts that a guess. I'm on iOS 9 w/no problems.

**Comment by Gigglequad** [comment]
Yes, I'm getting the same modifications error. Tivo still supports IOS7 as it works on my unmodded phone.

**Comment by Wildwillis** [comment]
I wanted to add to this issue as well because I'm a new TiVo Bolt owner.  I'm jailbroken on iOS 9.0.2 and I get the unauthorized modifications message on the latest version of the TiVo app (3.7.3) and xcon v42b2.  I also made sure SSH was disabled.

I tried downgrading to v3.53 of the TiVo app via App Admin but that didn't exactly work for me. I no longer get the unauthorized modifications message when trying to setup streaming in the Settings menu, but the app doesn't find any streaming devices when it searches for one. I'm guessing that version of the app came out before the TiVo Bolt did.

I tried running the command above in Terminal but didn't have any luck.  I also tried creating my own Flex patch but no luck there either. Really hoping for an xcon update to disable the jailbreak detection in this app. Thanks!

**Comment by ContextPerspective** [comment]
Here is how I got this to work with Roamio. Not sure about Bolt (don't own one) This is on 9.3.3 on both iPad Pro and iPhone 6S w/Roamio. This is with xCon 42b2

1. Remove all versions of Tivo
2. Load v.3.5.3 via IPAInstaller (You may need to play with install options, such as force install or Clean Install). Remove itunesmetadata option may error, if it does, remove that and try again.
3. Navigate to the TiVo app folder using Fileza. Remove iTunesMetadata.plist. This stops autoupdate.
4. Using Flex 3 Beta, download and enable all 3 patches, even if you don't plan to use those features
5. **You may not be able to stream, but you should be able to download.** Attempt to download a video. 
6. If you get the unauthorized modification error, reboot.
7. Try again. Same issue? Make sure  all running apps are closed, especially TiVo, then....

go into terminal...

at mobile$ - enter **su**
at Password: **Enter your SSH password**
at /var/mobile root# enter exactly this ----> **ldid -S /Library/MobileSubstrate/DynamicLibraries/xCon.dylib**
You won't see any confirmation, other than the cursor returning to /var/mobile root# 

You may need to repeat the terminal steps above a few times with reboot and without. Just make sure the tivo app is closed.
 
Also, you also need to occasionally rerun the command. It survives most reboots, but you may need to repeat the process.

**Comment by Westside-guy** [comment]
I'm having this same problem when trying to watch a show from my Bolt on a (32-bit) original iPad Mini running iOS 8.4.

**Comment by Wildwillis** [comment]
Thanks for the suggestions bsdteam.  I think the issue with 3.5.3 is that it pre-dates the TiVo Bolt.  I can downgrade to v3.5.3 of the app via App Admin and get into Streaming setup with xcon 42b2 installed without having to run through any of the other steps you listed.

During Streaming setup in the app however it doesn't find any streaming devices. I'd probably work just fine if I had a Roamio.  I think I'm dependent on an xcon update that bypasses jailbreak detection on a newer version of the TiVo app.

**Comment by Gigglequad** [comment]
Would someone mind uploading 3.5.3 somewhere? I'm unable to find it online. App Admin won't work on 7x IOS

**Comment by bmwrocks** [comment]
Running 9.3.3 I removed the current version of the Tivo app, installed AppAdmin then I downloaded 3.5.3 then installed Flex 3 and selected the Security and System Configuration aspects of the Tivo app and after activating the patch I signed into the Tivo app, setup streaming and I was streaming! I should mention that I had xCon 9.3.3 installed from the repo http://mohadu31.com and that was it. I just tried it without xCon and it stopped working.
